src/app/applications/datatarget/datatarget-detail/datatarget-detail.component.ts

```diff
@@ -19,79 +15,50 @@ import { faArrowsAltH } from '@fortawesome/free-solid-svg-icons';
 })
 export class DatatargetDetailComponent implements OnInit, OnDestroy {
 
-    public datatargetSubscription: Subscription;
-    public datatarget: Datatarget;
-    public backButton: BackButton = { label: '', routerLink: '/datatarget-list' };
-    public dataTargetRelations: PayloadDeviceDatatargetGetByDataTarget[];
-    private deleteDialogSubscription: Subscription;
-    public dropdownButton: DropdownButton;
-    arrowsAltH = faArrowsAltH;
-    private applicationName: string;
+    @ViewChild(DatatargetDetailTypeSelectorDirective, {static: true}) adHost!: DatatargetDetailTypeSelectorDirective;
 
-    constructor(
-        private route: ActivatedRoute,
-        private deleteDialogService: DeleteDialogService,
-        private location: Location,
-        private datatargetRelationServicer: PayloadDeviceDatatargetService,
+    private applicationName: string;
+    public datatarget: Datatarget;
+    private datatargetType: DataTargetType;
+        
+    constructor(private componentFactoryResolver: ComponentFactoryResolver,  
         private datatargetService: DatatargetService,
-        public translate: TranslateService) { }
+        private route: ActivatedRoute,
+        private datatargetTypesService: DatatargetTypesServiceService
+        ) { }
+   
 
-    ngOnInit(): void {
-        const id: number = +this.route.snapshot.paramMap.get('datatargetId');
-        this.applicationName = this.route.snapshot.paramMap.get('name');
-        if (id) {
-            this.getDatatarget(id);
-            this.getDatatargetRelations(id);
-            this.dropdownButton = {
-                label: '',
-                editRouterLink: '../../datatarget-edit/' + id,
-                isErasable: true,
-            }
-        }
-        this.translate.get(['NAV.MY-DATATARGET', 'DATATARGET.SHOW-OPTIONS'])
-            .subscribe(translations => {
-                this.backButton.label = translations['NAV.MY-DATATARGET'];
-                this.dropdownButton.label = translations['DATATARGET.SHOW-OPTIONS']
-            });
-    }
+    loadComponent(componentType: Type<any>) {
+       
+        const viewContainerRef = this.adHost.viewContainerRef;
+    
+        viewContainerRef.clear();
+        const factory = this.componentFactoryResolver.resolveComponentFactory(componentType);
+        const componentRef = viewContainerRef.createComponent<DatatargetDetail>(factory)
```

Comment:
Missing Semicolon (and unused componentref)

Reply:
fixed